addons/purchase_stock/models/stock.py

```diff
@@ -163,10 +164,84 @@ def _prepare_move_default_values(self, return_line, new_picking):
 class Orderpoint(models.Model):
     _inherit = "stock.warehouse.orderpoint"
 
+    show_supplier = fields.Boolean('Show supplier column', compute='_compute_show_suppplier')
+    supplier_id = fields.Many2one(
+        'product.supplierinfo', string='Vendor',
+        domain="['|', ('product_id', '=', product_id), '&', ('product_id', '=', False), ('product_tmpl_id', '=', product_tmpl_id)]")
+
+    def _get_default_route_id(self):
+        if self.product_id._prepare_sellers():
+            route_id = self.env['stock.rule'].search([
+                ('action', '=', 'buy')
+            ]).route_id
```

Comment:
to batch + move the lookup of buy rule out of the loop (and find all in one call to search)